### ISSUE TITLE
Use Python code for 3dMEPFM

### ISSUE THREAD

**Issue by tsalo** [ME-ICA/multi-echo-data-analysis#3]
@eurunuela is working on a Python implementation of 3dMEPFM, so we can use that instead of AFNI (which is harder to install for a Jupyter Book) once it's available.

# Original content

I'm thinking that we should use the [3dMEPFM R code](https://github.com/afni/afni/blob/5bc47685fc730cd2a975c3d427c05c78ad71c260/src/R_scripts/3dMEPFM.R) directly, rather than installing AFNI. To that end, we'll need to include R dependencies and have the relevant notebook use an R kernel instead of Python.

**Comment by tsalo** [comment]
@eurunuela any update on this?

**Comment by eurunuela** [comment]
> @eurunuela any update on this?

Working on improving one of the versions. Once I'm happy with it, I'll make the repo public and I'll add a Zenodo DOI.

**Comment by tsalo** [comment]
Thanks @eurunuela!

**Comment by eurunuela** [comment]
This is not the PFM version I referred to on my previous comments but should be good enough to use as an equivalent to 3dMEPFM: https://github.com/eurunuela/pySPFM

**Comment by tsalo** [comment]
@eurunuela the most recent release doesn't have all of the necessary dependencies listed, but I think you fixed that problem in a later commit. Can you make a new release?

**Comment by eurunuela** [comment]
I just made a new release. Have you tried running it?

I must warn you that I have only tested some parts of the code, but not everything. I am currently on the process of moving all the code I've written in the past 4 years into 3 proper Python packages. So some things may be broken atm.

**Comment by eurunuela** [comment]
I see. I'll have a look at it tonight or tomorrow and get back to you.

**Comment by tsalo** [comment]
No rush. I just had a few minutes and decided to add the dependency. I can totally wait before circling back to this.

**Comment by eurunuela** [comment]
> No rush. I just had a few minutes and decided to add the dependency. I can totally wait before circling back to this.

I just deployed to PyPi from my computer. I don't know why the Action failed.

**Comment by tsalo** [comment]
It's now installing correctly, but the new problem is the AFNI dependency. I've opened an issue in the `pyspfm` repo about it.

**Comment by tsalo** [comment]
I could probably use the AFNI+miniconda Docker image you were using for pyspfm's CI, but I'd definitely prefer to use pure Python in the future, so I'm hoping there will be a good way to update BRIK/HEAD files in the near future. 🤞

**Comment by eurunuela** [comment]
> I could probably use the AFNI+miniconda Docker image you were using for pyspfm's CI, but I'd definitely prefer to use pure Python in the future, so I'm hoping there will be a good way to update BRIK/HEAD files in the near future. 🤞

I agree it would be nice to have a pure Python version. But right now, adding this feature to `nibabel` is not the main priority for me. If someone develops it, I will happily include it to `pySPFM`.

**Comment by tsalo** [comment]
Per @eurunuela in this week's tedana devs call, the pure Python version of `pySPFM` should be good to go! I will try using it soon.
